javascript/i18n/phonenumbers/phonenumberutil.js

```diff
@@ -743,47 +743,122 @@ i18n.phonenumbers.PhoneNumberUtil.VALID_PHONE_NUMBER_ =
+ // button with the extension number following.
+ /** @type {string} */
+ var autoDiallingAndExtLabelsFound = "(?:,{2}|;)";
+ /** @type {string} */
+ var optionalExtnSuffix = "#?";
+
+ /** @type {string} */
+ var rfcExtn = i18n.phonenumbers.PhoneNumberUtil.RFC3966_EXTN_PREFIX_
+        + i18n.phonenumbers.PhoneNumberUtil.extnDigits_(extLimitAfterExplicitLabel);
+ /** @type {string} */
+ var explicitExtn = possibleSeparatorsBetweenNumberAndExtLabel + explicitExtLabels
+        + possibleCharsAfterExtLabel
+        + i18n.phonenumbers.PhoneNumberUtil.extnDigits_(extLimitAfterExplicitLabel)
+        + optionalExtnSuffix;
+ /** @type {string} */
+ var ambiguousExtn = possibleSeparatorsBetweenNumberAndExtLabel + ambiguousExtLabels
+        + possibleCharsAfterExtLabel
+	+ i18n.phonenumbers.PhoneNumberUtil.extnDigits_(extLimitAfterAmbiguousChar)
+	+ optionalExtnSuffix;
+ /** @type {string} */
+ var americanStyleExtnWithSuffix = ambiguousSeparator
+	+ i18n.phonenumbers.PhoneNumberUtil.extnDigits_(extLimitWhenNotSure) + "#";
+ // This is same as possibleSeparatorsBetweenNumberAndExtLabel, but not matching
+ // comma as extension label may have it.
+ /** @type {string} */
+ var possibleSeparatorsNumberExtLabelNoComma = "[ \u00A0\\t]*";
+
+ /** @type {string} */
+ var autoDiallingExtn = possibleSeparatorsNumberExtLabelNoComma
+        + autoDiallingAndExtLabelsFound + possibleCharsAfterExtLabel
+        + i18n.phonenumbers.PhoneNumberUtil.extnDigits_(extLimitAfterLikelyLabel)
+	+ optionalExtnSuffix;
+ /** @type {string} */
+ var onlyCommasExtn = possibleSeparatorsNumberExtLabelNoComma
+       + "(?:,)+" + possibleCharsAfterExtLabel
+       + i18n.phonenumbers.PhoneNumberUtil.extnDigits_(extLimitAfterAmbiguousChar)
+       + optionalExtnSuffix;
+
+ // The first covers RFC 3966 format, where the extension is added using ";ext=".
+ // The second more generic where extension is mentioned with explicit labels like
+ // "ext:". In both the above cases we allow more numbers in extension than any
+ // other extension labels. The third one captures when single character extension
+ // labels or less commonly used labels are used. In such cases we capture fewer
+ // extension digits in order to reduce the chance of falsely interpreting two
+ // numbers beside each other as a number + extension. The fourth one covers the
+ // special case of American numbers where the extension is written with a hash
+ // at the end, such as "- 503#". The fifth one exclusive for extension autodialling
+ // formats which are used when dialling and in this case we accept longer extensions.
+ // The last one is more liberal on number of commas that acts as extension labels,
+ // so we have strict cap on number of digits in such extensions.
+ /** @type {string} */
+ var extensionPattern =
+     rfcExtn + "|"
+     + explicitExtn + "|"
+     + ambiguousExtn + "|"
+     + americanStyleExtnWithSuffix + "|"
+     + autoDiallingExtn + "|"
+     + onlyCommasExtn;
+
+ return extensionPattern;
```

Comment:
I wouldn't bother constructing it separately, you can just do return rfcExtn + ....

Reply:
Done.